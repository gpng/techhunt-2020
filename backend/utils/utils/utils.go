package utils

import (
	"log"
	"mime/multipart"
	"net/http"
	"runtime"

	"golang.org/x/crypto/bcrypt"
)

// ContextKey is the unique key that represents a context value
type ContextKey string

func (c ContextKey) String() string {
	return "context key " + string(c)
}

// HashPassword will generate a bcrypt hash
func HashPassword(password string) (string, error) {
	bytes, err := bcrypt.GenerateFromPassword([]byte(password), 14)
	return string(bytes), err
}

// CheckPasswordHash will compare bcrypt hash with password
func CheckPasswordHash(password, hash string) bool {
	err := bcrypt.CompareHashAndPassword([]byte(hash), []byte(password))
	return err == nil
}

// GetMimeType from file
func GetMimeType(file multipart.File) (string, error) {
	fileHeader := make([]byte, 512)
	if _, err := file.Read(fileHeader); err != nil {
		return "", err
	}

	if _, err := file.Seek(0, 0); err != nil {
		return "", err
	}

	return http.DetectContentType(fileHeader), nil
}

// NewTrue initialises a true bool in memory and returns the address
// for use in struct bool pointers
func NewTrue() *bool {
	b := true
	return &b
}

// IsStringInSlice to check if string present in slice of strings
func IsStringInSlice(str string, list []string) bool {
	for _, s := range list {
		if s == str {
			return true
		}
	}
	return false
}

// LogError to console
func LogError(err error) {
	// get function caller name
	pc, file, line, ok := runtime.Caller(1)
	details := runtime.FuncForPC(pc)
	if ok && details != nil {
		log.Printf("%s:%d:%s failed with error: %v", file, line, details.Name(), err)
	}
}
