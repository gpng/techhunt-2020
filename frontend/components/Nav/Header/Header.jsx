import React from 'react';
// components
import Avatar from '../../Profile/Avatar';
// constants
import { COLORS } from '../../../constants/styles';
import DrawerSlider from '../Drawer/DrawerSlider';

const Header = () => {
  return (
    <nav className="header-root">
      <DrawerSlider />
      <Avatar width="2rem" />
      <style jsx>{`
        .header-root {
          background: ${COLORS.BACKGROUND_DARK_PRIMARY};
          height: 100%;
          width: 100%;
          display: flex;
          justify-content: space-between;
          align-items: center;
          padding: 0 1rem;
        }
      `}</style>
    </nav>
  );
};

export default Header;
