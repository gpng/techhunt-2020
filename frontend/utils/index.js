/**
 * Takes a promise, and returns then promised results in format [err, response]
 * @param { Promise } promise
 * @param { Object= } errorExt - Additional Information you can pass to the err object
 * @return { Promise }
 */
export const to = (promise, errorExt) => {
  return promise
    .then((data) => [null, data])
    .catch((err) => {
      if (errorExt) {
        Object.assign(err, errorExt);
      }

      return [err, undefined];
    });
};

export default to;
