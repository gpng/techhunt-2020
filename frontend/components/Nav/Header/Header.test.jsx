import React from 'react';
import { cleanup } from '@testing-library/react';
import Header from './Header';
import { renderWithContext } from '../../../utils/tests';

afterEach(cleanup);

it('should render', () => {
  renderWithContext(<Header />);
});

it('should match default snapshot', () => {
  const { asFragment } = renderWithContext(<Header />);
  expect(asFragment()).toMatchSnapshot();
});
