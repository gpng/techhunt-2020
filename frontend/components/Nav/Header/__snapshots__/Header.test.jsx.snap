// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match default snapshot 1`] = `
<DocumentFragment>
  <nav
    class="jsx-3120782651 header-root"
  >
    <div
      class="jsx-2616149830 drawer-slider-root"
    >
      <button
        class="jsx-2616149830 button-menu"
        data-testid="button-menu"
        type="button"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            fill="#f0f0f0"
          />
        </svg>
      </button>
      <div
        class="jsx-2616149830 drawer-slider-container"
      >
        <aside
          class="jsx-3606731684 drawer-root"
        >
          <div
            class="jsx-1818524733 avatar-root"
          />
          <div
            class="jsx-3606731684 button-container"
          >
            <button
              class="jsx-3606731684 button-function"
              data-testid="button-upload"
              type="button"
            >
              Upload CSV
            </button>
          </div>
        </aside>
      </div>
    </div>
    <div
      class="jsx-2124352701 avatar-root"
    />
  </nav>
</DocumentFragment>
`;
